MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -2309,13 +2329,26 @@ static uint32_t mvkGetEntryProperty(io_registry_entry_t entry, CFStringRef prope
 		return;
 	}
 
+	// kIOMasterPortDefault was deprecated and replaced by kIOMainPortDefault in macOS 12,
+	// and was removed altogether from MacCatalyst in macOS 14 beta.
+	// Both are documented to resolve to MACH_PORT_NULL.
+#if MVK_XCODE_13
+#	if MVK_MACCAT
+	const mach_port_t kIOMainPortDefaultMVK = mvkOSVersionIsAtLeast(12.0) ? kIOMainPortDefault : MACH_PORT_NULL;
+#	else
+	const mach_port_t kIOMainPortDefaultMVK = mvkOSVersionIsAtLeast(12.0) ? kIOMainPortDefault : kIOMasterPortDefault;
+#	endif
+#else
+	const mach_port_t kIOMainPortDefaultMVK = kIOMasterPortDefault;
+#endif
+
```

Comment:
Why don't we just use `MACH_PORT_NULL` everywhere? That's been the behavior of IOKit since long before Metal was introduced--since at least 10.4, in fact, and I think maybe even earlier.

Reply:
Mostly I was just timid about changing more than the minimum required, based on old Apple code comments.

But okay...updated now.